### DISCUSSION TITLE
SunbirdED installation on Oracle Cloud

### DISCUSSION THREAD

**Discussion by Munishdev97** [Sunbird-Ed/Community#645] [Installation] [answered]
For creating Kubernetes cluster in OCI there in no option to choose 1.24.x version which is mentioned in the documentation as It's not currently supported.

**Comment by gandham-santhosh** [comment]
@Munishdev97 We have verified it with 1.24.x and that is why it is mentioned in the documentation. You can go ahead and provision the next available version and share your findings

**Reply by Munishdev97** [reply]
Thanks for your update,
We'll go with the next version that has no end of life date and update our findings incase of any incompatibility.

we have some list of queries about OCI setup
1. In OKE, oracle Linux is only supported, Shall we use the same?. Is there any dependency on ubuntu within Kubernetes workload?
2. Which ubuntu version should we use for VM creation?

**Reply by santhosh-tg** [reply]
Hi @Munishdev97 
1. For OKE - yes, you can only choose Oracle Linux - better go with version **Oracle Linux 7.9 - 2023.07.31-1**  as there is an issue with Oracle 8 Linux and containers with side cars, which needs some additional configurations  to be done .
2. We have verified on Ubuntu 18.04 but some of our adopters have successfully installed it on Ubuntu 20.04 as well.

Cc: @gandham-santhosh @ddevadat

**Reply by Munishdev97** [reply]
In Jenkins server setup script, Jenkins version is 2.346.3. While using this version, many suggested plugins has failed to install due to incompatibility. So for that can we upgrade the Jenkins to next available version that supports the required plugins.

**Reply by Munishdev97** [reply]
Hi @gandham-santhosh @santhosh-tg

1. We have updated Jenkins to latest version 2.414.3 LTS and plugins are working fine now. Is there any other dependencies we have to check.
2. For the Kubernetes, we have chosen the 1.27.2 version and node OS version Oracle Linux 7.9 - 2023.07.31-1, we'll share our findings if there is any specific to this setup.

please let us know if there is any issues in choosing the above versions for Jenkins and Kubernetes.

**Comment by Munishdev97** [comment]
Hi @gandham-santhosh @santhosh-tg

Pod Security policy has been removed in Kubernetes 1.25 version onwards. Is there any other branch being worked on for the replacement with Pod Security admission controller.

So far from what we have checked there is dependency to Pod Security policy in monitoring, logging, fusionauth pipelines under deploy. Are there any other pipelines apart from these.

For the other deprecated resources from 1.24 onwards like pod disruption budget, cronjobs there is only change in apiVersion, which should not be a major issue. Kindly advise on how we can proceed with Pod Security Policy.

**Errors:  unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"**

**Reply by gandham-santhosh** [reply, marked as answer]
@Munishdev97 pod security policy changes are done here https://github.com/project-sunbird/sunbird-devops/pull/3926 . This should not have an impact on the functional application services. So I am guessing this will not be the blocker to set up 
You can leverage above PR to deploy the changes depending on Pod security policy and let us know

**Comment by anisha147** [comment]
Hi @Munishdev97 

We hope you were able to resolve the installation issues. If yes, could you please close this thread?

cc: @gandham-santhosh [@mohitga](https://github.com/mohitga)

**Reply by Munishdev97** [reply]
Hello @anisha147 ,
we used the release-7.0.0 as mentioned by @gandham-santhosh for monitoring namespace and it is working as expected. Closing the thread.
